### PR TITLE
devicestate: fix serial acquire when custom SSL certificates are needed

### PR DESCRIPTION
**Pull request by mvo5**
This PR fixes the issue that getting the serial fails in confusing ways when a custom SSL certificate is needed to connect
to get the serial. The error message in this case is:
```
handlers_serial.go:659: Proxy store does not support custom serial vault; ignoring the proxy
```
which is not at all true.

The root cause for the bug is that the used http client does
set `ExtraSSLCerts`. However there are more issues with the
code, e.g. that `newEnoughProxy` does not return an error but
instead just debug lots it and does nothing less.

The initial commit is a minimal fix, then a unit test change to ensure the MockDeviceService uses https with custom certificates and then a refactor of `newEnoughProxy` to ensure that errors are properly surfaced.

One open question is if `httputil.NewHTTPClient` should by default set: `ExtraSSLCerts: &httputil.ExtraSSLCertsFromDir{Dir: dirs.SnapdStoreSSLCertsDir}` unless disabled to avoid similar issues in the future. All places in our code that call `httputil.NewHTTPClient()` set this option anyway AFAICT.

### CODE REVIEW

**Comment by pedronis** [comment]
> One open question is if httputil.NewHTTPClient should by default set: ExtraSSLCerts: &httputil.ExtraSSLCertsFromDir{Dir: dirs.SnapdStoreSSLCertsDir} unless disabled to avoid similar issues in the future. All places in our code that call httputil.NewHTTPClient() set this option anyway AFAICT.

this involve importing dirs from httputil tough

**Review by pedronis — APPROVED** [review]
the fix looks right, comment about the MockDeviceService changes though

File: overlord/devicestate/devicestatetest/devicesvc.go [hunk 1]
```diff
@@ -252,4 +257,24 @@ func MockDeviceService(c *C, bhv *DeviceServiceBehavior) *httptest.Server {
 			}
 		}
 	}))
+	// write custom test-server TLS certificates to
+	// dirs.SnapdStoreSSLCertsDir
+	err := os.MkdirAll(dirs.SnapdStoreSSLCertsDir, 0755)
```
[the remark this answers is not in the document]
Comment (mvo5) [reply overlord/devicestate/devicestatetest/devicesvc.go]: Thanks, I updated the code now to follow this pattern

[the remark this answers is not in the document]
Comment (mvo5) [reply overlord/devicestate/devicestatetest/devicesvc.go]: Thinking about it, I could create a `storetest.MakeExtraSnapdSSLCert(name, pemEncodedExtraCerts)` to consolidate the helper into a single place? Wdyt?

[the remark this answers is not in the document]
Comment (pedronis) [reply overlord/devicestate/devicestatetest/devicesvc.go]: not against, but do we have other test places that could use this helper?

**Review by MiguelPires — APPROVED** [review]

**Pull request merged**